src/commands/prompts/finish.ts

```diff
@@ -70,23 +51,52 @@ const askTimeRemaining = async (tpEntity: any, timeEntry: HarvestTimeEntry) => {
     return parseFloat(timeRemaining);
 };
 
-export const askFinishDetails = async (apiProvider: ApiProvider, date: string) => {
-    const harvestApi = apiProvider.getHarvestApi();
+const getTimeEntries = async (harvestApi: HarvestApi, date: string, all: boolean) => {
+    const entries = await harvestApi.getTimeEntries(date);
 
-    const timeEntry = await askTimeEntry(harvestApi, date);
+    const unfinished = getUnfinishedTimeEntries(entries);
 
-    if (timeEntry === null) {
-        return null;
+    if (unfinished.length === 0) {
+        log.info(`There are no unfinished time entries on ${date}.`);
+        return [];
     }
 
+    if (all) {
+        return unfinished;
+    }
+
+    const prompts = unfinished.map(getTimeEntryPrompt);
+
+    const { timeEntry } = await inquirer.prompt<{ timeEntry: HarvestTimeEntry }>({
+        name: "timeEntry",
+        message: "Which timer would you like to finish?",
+        type: "list",
+        choices: prompts
+    });
+
+    return [ timeEntry ];
+};
+
+export const askFinishDetails = async (apiProvider: ApiProvider, date: string, all: boolean) => {
+    const harvestApi = apiProvider.getHarvestApi();
+
+    const timeEntries = await getTimeEntries(harvestApi, date, all);
+
     const targetprocessApi = apiProvider.getTargetprocessApi();
-    const tpEntity = await getTargetprocessEntity(targetprocessApi, timeEntry.notes.entity.id);
 
-    const timeRemaining = await askTimeRemaining(tpEntity, timeEntry);
+    const finishDetails = [];
 
-    return {
-        tpEntity,
-        timeEntry,
-        timeRemaining
-    };
+    for (const timeEntry of timeEntries) {
+        const tpEntity = await getTargetprocessEntity(targetprocessApi, timeEntry.notes.entity.id);
+
+        const timeRemaining = await askTimeRemaining(tpEntity, timeEntry);
+
+        finishDetails.push({
```

Comment:
I assume the targetProcess Api deals nicely with timeRemaining being `NaN`?

Reply:
`askTimeRemaining` checks that `isNaN` is false before parsing